config/main.py

```diff
@@ -33,6 +35,70 @@ def run_command(command, display_cmd=False, ignore_error=False):
     if proc.returncode != 0 and not ignore_error:
         sys.exit(proc.returncode)
 
+def interface_alias(interface_name):
+    """Return default interface name if alias name is given as argument
+    """
+
+    cmd = 'sonic-cfggen -d --var-json "PORT"'
+    p = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE)
+
+    port_dict = json.loads(p.stdout.read())
+
+    if interface_name is not None:
+        for port_name in natsorted(port_dict.keys()):
+            if interface_name == port_dict[port_name]['alias']:
+                return port_name
+        print "Invalid interface {}".format(interface_name)
+        sys.exit(0)
+
+def interface_name_to_alias(interface_name):
+    """Return alias interface name if default name is given as argument
+    """
+
+    cmd = 'sonic-cfggen -d --var-json "PORT"'
+    p = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE)
+
+    port_dict = json.loads(p.stdout.read())
+
+    if interface_name is not None:
+        for port_name in natsorted(port_dict.keys()):
+            if interface_name == port_name:
+                return port_dict[port_name]['alias']
+        print "Invalid interface {}".format(interface_name)
+        sys.exit(0)
+
+
+
+def set_interface_mode(mode):
+    """Modify IFMODE env variable in user .bashrc
+    """
+    sudo_user = os.getenv('SUDO_USER')
+    user = os.getenv('USER')
+    set_mode = "IFMODE={}".format(mode)
+
+    if not sudo_user:
+        bashrc = "/{}/.bashrc".format(user)
```

Comment:
The .bashrc path should always be `"/home/{}/.bashrc"`, whether or not the user has called the command with sudo.

I also believe you can simplify lines 75-82 as follows:

```
bashrc_ifmode_line = "IFMODE={}".format(mode)

user = os.getenv('SUDO_USER')

if not user:
    user = os.getenv('USER')

bashrc = "/home/{}/.bashrc".format(user)
...
```

Reply:
- What if root user try to do the change the .bashrc will be in /root/.bashrc. 
- Do we need to adress this scenario?

Reply:
We never want to modify the .bashrc of the root user. This is the entire point of first checking to see if `SUDO_USER` is not empty. If SUDO_USER is not empty, we want to modify *that* user's .bashrc. Similarly, we want to check the environment variables of *that* user to determine the mode to use.